src/typechecker.rs

```diff
@@ -50,9 +51,38 @@ pub fn is_integer(type_id: TypeId) -> bool {
     }
 }
 
+pub fn is_signed(type_id: TypeId) -> bool {
+    match type_id {
+        USIZE_TYPE_ID | U8_TYPE_ID | U16_TYPE_ID | U32_TYPE_ID | U64_TYPE_ID => false,
```

Comment:
This is not sufficient:

> https://doc.rust-lang.org/std/os/raw/type.c_char.html
> On modern architectures this type will always be either `i8` or `u8`, as they use byte-addresses memory with 8-bit bytes.

so we need to check for that.

Reply:
That is indeed a good point. From what I can see in the documentation, it looks like the only way to check if c_char is signed is by checking if `c_char::MIN` is smaller than zero. This seems like a hacky solution but I don't see another way to check for that.